src/js/components/award/contract/ContractDetails.jsx

```diff
@@ -4,185 +4,26 @@
-                if (popPlace !== '') {
-                    popPlace += '\n';
-                }
-                popPlace +=
-            `Congressional District: ${award.pop_state_code}-${award.pop_congressional_district}`;
-            }
-        }
-        else if (award.pop_country_code !== 'USA') {
-            popPlace = `${award.pop_country}`;
-        }
-
-        if (popPlace === '') {
-            popPlace = 'Not available';
-        }
-
-        return popPlace;
-    }
-
-    prepareValues(award) {
-        let yearRangeTotal = "";
-        let monthRangeTotal = "";
-        let description = null;
-
-        // Date Range
-        const formattedStartDate = award.period_of_performance_start_date;
-        const formattedEndDate = this.props.endDate;
-
-        const startDate = moment(formattedStartDate, 'M/D/YYYY');
-        const endDate = moment(formattedEndDate, 'M/D/YYYY');
-
-        const duration = moment.duration(endDate.diff(startDate));
-        const years = duration.years();
-        const months = duration.months();
-
-        let popDate = "Not Available";
-        if (!isNaN(years)) {
-            if (months > 0) {
-                if (months === 1) {
-                    monthRangeTotal = `${months} month`;
-                }
-                else {
-                    monthRangeTotal = `${months} months`;
-                }
-            }
-
-            if (years > 0) {
-                if (years === 1) {
-                    yearRangeTotal = `${years} year`;
-                }
-                else {
-                    yearRangeTotal = `${years} years`;
-                }
-            }
-
-            let timeRange = '';
-            if (monthRangeTotal && yearRangeTotal) {
-                timeRange = `(${yearRangeTotal}, ${monthRangeTotal})`;
-            }
-            else if (monthRangeTotal) {
-                timeRange = `(${monthRangeTotal})`;
-            }
-            else if (yearRangeTotal) {
-                timeRange = `(${yearRangeTotal})`;
-            }
-
-            popDate = `${formattedStartDate} - ${formattedEndDate} ${timeRange}`;
-        }
-
-        if (award.description) {
-            description = award.description;
-        }
-        else {
-            description = "Not Available";
-        }
-
-        // Award Type
-        const awardType = this.props.awardType;
-
-        // Pricing
-        let pricing = "Not Available";
-        if (award.type_of_contract_pricing_description) {
-            pricing = award.type_of_contract_pricing_description;
-        }
-
-        // char count
-        let seeMore = false;
-        if (award.description.length > this.props.maxChars) {
-            seeMore = true;
+    render() {
+        const award = this.props.selectedAward;
+        let awardType = award.awardType;
+        if (award.category === 'idv') {
+            awardType = idvAwardTypes[this.props.selectedAward.awardType];
```

Comment:
This logic is broken for IDVs because `awardType` was remapped to `latest_transaction.contract_data.idv_type_description` instead of `latest_transaction.contract_data. idv_type`

Reply:
More generally, because `DetailRow` performs a length check on the string, we must ensure that whatever we pass into `DetailRow` has a length value (not `null` or `undefined`) or the page will fail to render.

Reply:
Look at `#/award/637`